hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/GetTokenExpiryInfoPrecompile.java

```diff
@@ -67,8 +67,9 @@ public Builder body(final Bytes input, final UnaryOperator<byte[]> aliasResolver
 
     @Override
     public Bytes getSuccessResultFor(final ExpirableTxnRecord.Builder childRecord) {
-        validateTrue(stateView.tokenExists(tokenId), ResponseCodeEnum.INVALID_TOKEN_ID);
-        final var tokenInfo = stateView.infoForToken(tokenId).orElse(null);
+        validateTrue(ledgers.tokens().contains(tokenId), ResponseCodeEnum.INVALID_TOKEN_ID);
```

Comment:
Having this check also in the beginning duplicates it. I think we can leave the check only for `tokenInfo`.